scripts/screeps-private-smoke-harness.py

```
@@ -0,0 +1,639 @@
+    except OSError:
+        pass
+    if existing == content:
+        return False
+    path.parent.mkdir(parents=True, exist_ok=True)
+    path.write_text(content, encoding="utf-8")
+    return True
+
+
+def download_map(map_path: Path, timeout: int) -> str:
+    """Download the pinned map JSON into the ignored runtime cache if possible."""
+    request = urllib.request.Request(MAP_URL, headers={"User-Agent": "screeps-private-smoke-harness/1.0"})
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)